src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,271 @@
+    unsigned int num_segments;
+    // The maximum acceleration allowed at any
+    //  point along the trajectory. This factor limits the maximum delta-velocity
+    // and
+    //  also the max speed around curves due to centripetal acceleration [m/s^2]
+    float max_allowable_acceleration;
+    // The maximum speed allowable at any point along the trajectory```
+    float max_allowable_speed;
+    // The initial speed at the start of the trajectory [m/s]
+    float initial_speed;
+    // The final speed at the end of the trajectory [m/s]
+    float final_speed;
+} FirmwareRobotPathParameters_t;
+
+/**
+ * Returns a planned trajectory with the list of guarantees based on the assumptions below
+ *
+ *  Key assumptions & guarantees of this planner are:
+ *  - Trajectories are time-optimal assuming INFINITE JERK capability of the robot
+ *
+ *  - No speed on the trajectory is larger than the 'max_allowable_speed' parameter
+ *
+ *  - No acceleration value between points on the trajectory can be larger than the
+ * 'max_allowable_acceleration' input parameter
+ *
+ *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION then at no
+ * point on the path can the sum of centripetal and acceleration force be greater than
+ * 'max_allowable_acceleration'
+ *
+ *  Trajectory generation is done by assuming constant acceleration capability.
+ *      - The generator will assume maximum acceleration for the robot between each
+ * segment on the path. If using max acceleration breaches either the speed limit of the
+ * robot, or the centripetal acceleration limit defined by the curvature and robot speed -
+ * the planner will assume the max acceleration that will remain bellow this limit.
+ *
+ *     - To ensure the robot is capable of deccelerating to react to any sudden changes in
+ * the path, the trajectory is checked for backwards continuity
+ *
+ * NOTE: The following function is based on an algorithm described in
+ * http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf
+ *
+ * @pre path_parameters.num_segments > 2 and num_segments <=
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre path_parameters.max_allowable_acceleration > 0
+ * @pre path_parameters.max_allowable_speed > 0
+ * @pre path_parameters.init_speed >= 0
+ * @pre path_parameters.final_speed >= 0
+ * @pre The trajectory.trajectory_elements[] array is pre-allocated to handle up to the
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS limit
+ *
+ * @param path_parameters [in] The data structure including important path parameters as
+ * defined by the FirmwareRobotPathParameters struct
+ * @param trajectory [out] The planned trajectory that follows robot dynamics limitations
+ * with the appropriate guarantees and assumptions outlines above
+ *
+ * @return The outcome of the trajectory generation. Returns OK if trajectory is valid.
+ */
+TrajectoryPlannerGenerationStatus
+app_trajectory_planner_generateConstantArcLengthSegmentation(
```

Comment:
nit

Just noticed this function name doesn't quite match the equivalent `app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation`. This version is missing `...Trajectory` before `Segmentation`. Can we make these consistent?

Reply:
:+1: Removed the 'Segmentation' from both